DependencyInjection/Compiler/RepositoryAliasPass.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+namespace Doctrine\Bundle\DoctrineBundle\DependencyInjection\Compiler;
+
+use Doctrine\Common\Persistence\Mapping\Driver\MappingDriver;
+use Doctrine\ORM\Mapping\ClassMetadata;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\HttpKernel\Kernel;
+
+class RepositoryAliasPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (!$container->hasParameter('doctrine.entity_managers')) {
+            return;
+        }
+
+        $entityManagers = $container->getParameter('doctrine.entity_managers');
+        $customRepositories = [];
+
+        foreach ($entityManagers as $name => $serviceName) {
+            $metadataDriverService = sprintf('doctrine.orm.%s_metadata_driver', $name);
+
+            if (!$container->has($metadataDriverService)) {
+                continue;
+            }
+
+            /** @var MappingDriver $metadataDriver */
+            $metadataDriver = $container->get($metadataDriverService);
+            $entityClassNames = $metadataDriver->getAllClassNames();
+
+            foreach ($entityClassNames as $entityClassName) {
+                $classMetadata = new ClassMetadata($entityClassName);
+                $metadataDriver->loadMetadataForClass($entityClassName, $classMetadata);
+
+                if ($classMetadata->customRepositoryClassName) {
+                    $customRepositories[$classMetadata->customRepositoryClassName][] = [
+                        0 => $entityClassName,
+                        1 => $name,
+                    ];
+                }
+            }
+        }
+
+        $rootConflicts = [];
+        foreach ($customRepositories as $repositoryClass => $entities) {
+            $repoConflicts = $this->findConflictingServices($container, $repositoryClass);
+
+            if (count($repoConflicts)) {
+                $rootConflicts[$repositoryClass] = $repoConflicts;
```

Comment:
It looks like the `$repoConflicts` isn't actually used. This could just be set to a boolean (and `findConflictingServices` could also be renamed and return a boolean).